code/Chapter4/Exercise_solutions/b_multiple_test_p_instant.cpp

```diff
@@ -0,0 +1,94 @@
+#include <gtest/gtest.h>
+
+// Define your function to be tested here.
+int Sum(int a, int b) {
+    return a + b;
+}
+
+// function to multiply two numbers
+double Multiply(double a, double b) {
+    return a * b;
+}
+
+// Function to calculate a raised to the power of b (integer) without using pow function
+double Power(double a, int b) {
+    double result = 1;
+    for(int i = 0; i < b; i++) {
+        result *= a;
+    }
+    return result;
+}
+
+// Define a test fixture class
+class ParameterizedTest : public testing::TestWithParam<std::pair<int, int>> {
+};
+
+// Define the test case with the parameterized test
+TEST_P(ParameterizedTest, TestSum) {
+    // Get the parameter values
+    int a = GetParam().first;
+    int b = GetParam().second;
+
+    // Call your normal function
+    int result = Sum(a, b);
+
+    // Perform assertion
+    ASSERT_EQ(a + b, result);
+}
+
+// Define the test data
+INSTANTIATE_TEST_SUITE_P(Default, ParameterizedTest, testing::Values(
+    std::make_pair(1, 1),
+    std::make_pair(2, 3),
+    std::make_pair(-5, 10)
+));
+
+// Define the test case with the parameterized test for multiply function.
+TEST_P(ParameterizedTest, TestMultiply) {
+    // Get the parameter values
+    int a = GetParam().first;
+    int b = GetParam().second;
+
+    // Call your normal function
+    int result = Multiply(a, b);
+
+    // Perform assertion
+    ASSERT_EQ(a * b, result);
+}
+
+
+// Define a test fixture class
+class ParameterizedTest_Power : public testing::TestWithParam<std::tuple<double, int, double>> {
+};
+
+
+//Check if the power function works fine for different values of a and b
+TEST_P(ParameterizedTest_Power, TestPowerFun)
+{
+    // Get the parameter values
+    double a = std::get<0>(GetParam());
+    int b = std::get<1>(GetParam());
+    double answer = std::get<2>(GetParam());
+
+    // Call your normal function
+    double result = Power(a, b);
+
+    // Perform assertion
+    ASSERT_EQ(answer, result);
```

Comment:
```suggestion
    ASSERT_DOUBLE_EQ(answer, result);
```

Reply:
Accepted in b766b21ae4cc6be29dc85087806f946fe7294db8